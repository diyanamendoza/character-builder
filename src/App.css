@import url('https://fonts.googleapis.com/css2?family=Alegreya+Sans+SC:ital,wght@1,800&family=Oswald:wght@600&display=swap');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: 'Alegreya Sans SC', sans-serif;
  background-color: darkslategray;
  color: white;
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: calc(10px + 2vmin);
}

main {
  display: flex;
}
