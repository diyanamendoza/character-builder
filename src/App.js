/* eslint-disable react/jsx-no-comment-textnodes */
import React from 'react'
import './App.css'
import { useState } from 'react'
import Picker from './components/Picker/Picker.jsx'
import Text from './components/Text/Text.jsx'
import Character from './components/Character/Character.jsx'
import SpeechBubble from './components/SpeechBubble/SpeechBubble.jsx'

function App() {
  //container component
  const [head, setHead] = useState('deadpool')
  const [torso, setTorso] = useState('deadpool')
  const [feet, setFeet] = useState('deadpool')
  const [text, setText] = useState('Change me')
  const [mottos, setMottos] = useState([])

  const handleClick = () => {
    setMottos((prevState) => [...prevState, text])
  }

  return (
    <div className="App">
      <header className="App-header">
        <h1>CHARACTER BUILDER</h1>
        {/*presentational component*/}
        <Picker
          head={head}
          setHead={setHead}
          torso={torso}
          setTorso={setTorso}
          feet={feet}
          setFeet={setFeet}
        />
        {/*spread version: { ...{ head, setHead }}*/}
        <Text text={text} setText={setText} handleClick={handleClick} />
      </header>
      <main>
        <Character head={head} torso={torso} feet={feet} />
        <SpeechBubble mottos={mottos} />
      </main>
    </div>
  )
}

export default App
