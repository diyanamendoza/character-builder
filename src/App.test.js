import { render, screen } from '@testing-library/react';
import App from './App';

it('renders app including header and main', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();

  const header = screen.getByRole('banner');
  expect(header).toBeInTheDocument();

  const main = screen.getByRole('main');
  expect(main).toBeInTheDocument();
});
