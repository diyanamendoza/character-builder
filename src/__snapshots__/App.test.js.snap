// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app including header and main 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        CHARACTER BUILDER
      </h1>
      <div
        class="picker"
        style="display: flex; flex-direction: column; gap: 10px; margin-bottom: 10px;"
      >
        <label>
          Head
          <select>
            <option>
              hulk
            </option>
            <option>
              deadpool
            </option>
            <option>
              spiderman
            </option>
            <option>
              captainmarvel
            </option>
          </select>
        </label>
        <label>
          Torso
          <select>
            <option>
              hulk
            </option>
            <option>
              deadpool
            </option>
            <option>
              spiderman
            </option>
            <option>
              captainmarvel
            </option>
          </select>
        </label>
        <label>
          Feet
          <select>
            <option>
              hulk
            </option>
            <option>
              deadpool
            </option>
            <option>
              spiderman
            </option>
            <option>
              captainmarvel
            </option>
          </select>
        </label>
      </div>
      <div
        class="text"
      >
        <label>
          Motto
          <input
            style="margin-left: 10px; width: 200px;"
            value="Change me"
          />
        </label>
        <button
          type="button"
        >
          Submit
        </button>
      </div>
    </header>
    <main>
      <div
        class="character"
        style="display: flex; flex-direction: column; gap: 10px;"
      >
        <div
          class="head"
          style="background-image: url(./deadpool-head.png); background-size: cover; height: 180px; width: 320px; background-repeat: no-repeat;"
        />
        <div
          class="torso"
          style="background-image: url(./deadpool-torso.png); background-size: cover; height: 280px; width: 320px; background-repeat: no-repeat;"
        />
        <div
          class="feet"
          style="background-image: url(./deadpool-feet.png); background-size: cover; height: 240px; width: 320px; background-repeat: no-repeat;"
        />
      </div>
      <div
        class="speechbubble"
      />
    </main>
  </div>
</div>
`;
